File: dms-frontend/dms-frontend/src/types/training.ts
import type { ExpiryStatus, SearchParams } from "./common";

export interface Training {
  trainingId: number;
  driverId: number;
  driverName: string;
  trainingTypeName: string;
  dateCompleted: string;
  validUpto: string;
  trainerName?: string | null;
  status: ExpiryStatus;
}

export interface CreateTrainingRequest {
  driverId: number;
  trainingTypeId: number;
  dateCompleted: string;
  validUpto: string;
  trainerName?: string;
}

export interface UpdateTrainingRequest {
  trainingTypeId: number;
  dateCompleted: string;
  validUpto: string;
  trainerName?: string;
}

export interface TrainingSearchParams extends SearchParams {
  driverId?: number;
  status?: ExpiryStatus;
}

File: src/Driver.Domain/Entities/Training.cs
using DriverDms.Domain.Common;

namespace DriverDms.Domain.Entities;

public class Training : BaseEntity
{
    public int TrainingId { get; set; }
    public int DriverId { get; set; }
    public Driver? Driver { get; set; }
    public int TrainingTypeId { get; set; }
    public TrainingType? TrainingType { get; set; }
    public DateTime DateCompleted { get; set; }
    public DateTime ValidUpto { get; set; }
    public string? TrainerName { get; set; }

    // Same reasoning as License.IsExpired / MedicalRecord.IsExpired (design doc §6.3):
    // "expiring soon" needs the configurable reminder window, which depends on
    // SystemConfiguration - that logic lives in TrainingStatusResolver, not here.
    public bool IsExpired => ValidUpto < DateTime.UtcNow;
}
